framework/src/utils/PetscSupport.C

```diff
@@ -435,6 +435,10 @@ petscNonlinearConverged(SNES snes,
       *reason = SNES_DIVERGED_LINE_SEARCH;
 #endif
       break;
+
+    case MooseNonlinearConvergenceReason::DIVERGED_NL_RESIDUAL_PINGPONG:
+      *reason = SNES_DIVERGED_LINE_SEARCH;
```

Comment:
Hmm it would be nice if there was a `SNES_DIVERGED_USER`

Reply:
But beyond the scope of MOOSE

Reply:
agree, somebody should add `SNES_DIVERGED_USER `

Reply:
I was thinking `SNES_DIVERGED_MAX_IT`    or  `SNES_DIVERGED_LOCAL_MIN` would be better than `SNES_DIVERGED_LINE_SEARCH`

Either way, it is super confusing

Reply:
for me is the same, it would be nice to have an additional output line letting users know when this is triggered.
But I cannot use std::cout or std::cerr because it will fail prechecks. Don't know what `oss << ` does but even that is masked...

Reply:
Yes, it has nothing to do with `line search `.  `SNES_DIVERGED_LOCAL_MIN` can explain part of secret